pkg/vetter/util/mtlspolicy/authPolicy.go

```diff
@@ -122,85 +160,254 @@ func (ap *AuthPolicies) ByPort(s Service, port uint32) []*authv1alpha1.Policy {
 	return p
 }
 
-// This returns all the policies for a cluster that whose target is a port, regardless of service or namespace
-func (ap *AuthPolicies) GetClusterPortPolicies() []*authv1alpha1.Policy {
-	portPolicies := []*authv1alpha1.Policy{}
-	for _, polsByNamePortMap := range ap.port {
-		for _, polsByPortMap := range polsByNamePortMap {
-			for _, policies := range polsByPortMap {
-				portPolicies = append(portPolicies, policies...)
+// ForEachPolByPort takes a callback and applies it to a range of policies by port
+func (ap *AuthPolicies) ForEachPolByPort(s Service, cb func(s Service, port uint32, policies []*authv1alpha1.Policy)) error {
+
+	ns, ok := ap.port[s.Namespace]
+	if !ok {
+		return errors.New("No policies for namespace that specify a port")
+	}
+	n, ok := ns[s.Name]
+	if !ok {
+		return errors.New("No policies for service that specify a port")
+	}
+	for port, policies := range n {
+		cb(s, port, policies)
+		return nil
+	}
+	return nil
+}
+
+// getMTLSBool returns a bool and error from the 4 possible enum mTls states.
+// Mixed counts as enabled since it allows enabled traffic, but it returns an error in case the caller needs to know if the true status means it's enabled-only, or enabled in a way that allows other traffic.
```

Comment:
It doesn't look like any of the callers that I saw care about the error.  Maybe the callers of those callers care?

If I'm remembering, the goal here was to narrow the 4-state MTLSSetting into the old bool interface that some callers cared about, so those callers could keep using the old interface they like.

If that's what your goal is, *and* you're trying to also support other callers that want a more subtle interface (like this bool + error), maybe we should have two helper functions?